App.tsx

```diff
@@ -130,90 +131,106 @@ const App = () => {
-      }
-    }, [fontsLoaded]);
-
-    if (!fontsLoaded) {
-      // The splash screen keeps rendering while fonts are loading
-      return null;
-    }
-
     return (
-      <ClientContext.Provider value={contextValue}>
-        <QueryClientProvider client={queryClient}>
-          <NativeBaseProvider theme={theme}>
-            <SafeAreaProvider>
-              <NavigationContainer>
-                <View onLayout={onLayoutRootView} style={StyleSheet.absoluteFill}>
-                  <TabNavigator.Navigator
-                    initialRouteName="Home"
-                    screenOptions={({route}) => ({
-                      headerShown: false,
-                      tabBarIcon: ({color, size}) => {
-                        if (route.name === 'Home') {
-                          return <AntDesign name="search1" size={size} color={color} />;
-                        } else if (route.name === 'Observations') {
-                          return <AntDesign name="filetext1" size={size} color={color} />;
-                        } else if (route.name === 'Weather Data') {
-                          return <AntDesign name="barschart" size={size} color={color} />;
-                        } else if (route.name === 'Menu') {
-                          return <AntDesign name="bars" size={size} color={color} />;
-                        }
-                      },
-                    })}>
-                    <TabNavigator.Screen name="Home" initialParams={{center_id: avalancheCenterId, date: date}}>
-                      {state => HomeTabScreen(withParams(state, {center_id: avalancheCenterId, date: date}))}
-                    </TabNavigator.Screen>
-                    <TabNavigator.Screen name="Observations" initialParams={{center_id: avalancheCenterId, date: date}}>
-                      {state => ObservationsTabScreen(withParams(state, {center_id: avalancheCenterId, date: date}))}
-                    </TabNavigator.Screen>
-                    <TabNavigator.Screen name="Weather Data" initialParams={{center_id: avalancheCenterId, date: date}}>
-                      {state => TelemetryTabScreen(withParams(state, {center_id: avalancheCenterId, date: date}))}
-                    </TabNavigator.Screen>
-                    <TabNavigator.Screen name="Menu" initialParams={{center_id: avalancheCenterId}}>
-                      {() => MenuStackScreen(avalancheCenterId, setAvalancheCenterId, staging, setStaging)}
-                    </TabNavigator.Screen>
-                  </TabNavigator.Navigator>
-                </View>
-              </NavigationContainer>
-            </SafeAreaProvider>
-          </NativeBaseProvider>
-        </QueryClientProvider>
-      </ClientContext.Provider>
+      <QueryClientProvider client={queryClient}>
+        <AppUsingQueryClient />
+      </QueryClientProvider>
     );
   } catch (error) {
     Sentry.Native.captureException(error);
     throw error;
   }
 };
 
+const AppUsingQueryClient = () => {
```

Comment:
1. this name is kind of odd, since this component is not referring to a query client at all. Maybe this refactoring would make more sense to me if the ordering was inverted, i.e. BaseApp, AppWithClientContext (which wraps BaseApp and adds ClientContext), App (which wraps AppWithClientContext and adds query client)
1. it seems inconsistent to have `staging` be in this component's state (and have to be plumbed downward), while values derived from `staging` are in context 🤔

Reply:
Your suggestions in 1 sound good to me!

Re: 2 - while the changes in this PR lay that fact bare, they don't actually introduce that inversion of control. I agree with you that it's odd that a sub-item (`Menu`) owns showing the state and mutating it, and the entire app consumes the outcome of that via the context, but I'm not sure if there's a logical inconsistency there. Do you have a suggestion for how else we might approach this? Throwing a larger context or redux at the problem IMHO just puts some syntactic sugar over the thing, without really changing the data ownership or data use flows.